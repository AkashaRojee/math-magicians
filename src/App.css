.calculator {
  display: grid;
  grid-template-rows: auto auto;
  width: max-content;
  margin: auto;
}

.display {
  justify-self: flex-end;
}

.keyboard {
  display: grid;
  grid-template-columns: repeat(4, 50px);
}

.span-two-col {
  grid-column: span 2;
}