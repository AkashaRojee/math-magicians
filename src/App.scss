* {
  margin: 0;
  padding: 0;
}

.bg-orange {
  background-color: orange;
}

%bg-grey {
  background-color: grey;
}

%grid {
  display: grid;
  color: white;
}

.calculator {
  @extend %grid;

  grid-template-rows: 50px auto;
  width: max-content;
  margin: auto;

  .display {
    @extend %bg-grey;

    text-align: right;
    padding: 15px;
  }

  .keyboard {
    @extend %grid;

    grid-template-columns: repeat(4, 100px);
    grid-template-rows: repeat(5, 50px);

    .span-two-col {
      grid-column: span 2;
    }
  }
}
