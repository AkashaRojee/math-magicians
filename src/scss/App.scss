@import 'layout.scss';
@import 'utilities.scss';
@import url('https://fonts.googleapis.com/css2?family=Spartan:wght@700&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  color: $yellow;
  font-family: 'Spartan', sans-serif;
}

body {
  @extend %p-20;
  height: 100vh;
  background-color: $violet-dark;
}

/*
  ========================================
  Header
  ========================================
*/

.header {
  @extend %flex-row;
  @extend %justify-between;
  @extend %align-center;
  @extend %bg-basic;

  nav {
    @extend %flex-row;
    gap: 25px;

    a {
      color: inherit;
      position: relative;

      & + a {
        &:before {
          content: '|';
          pointer-events: none;
          position: absolute;
          left: -15px;
          vertical-align: -50%;
        }
      }
    }
  }
}

/*
  ========================================
  Main / Routes
  ========================================
*/

main {
  @extend %p-x-15;
}

.home-route {
  @extend %flex-column;
  @extend %m-t-50;
  gap: 50px;

  p + p {
    &:before {
      content: '\a';
      white-space: pre;
    }
  }  
}

.calculator-route {
  @extend %flex-row;
  @extend %justify-between;
  @extend %m-y-50;
}

.quote-route {
  @extend %flex-row;
  @extend %align-center;
  @extend %m-x-25;
  height: calc(100vh - 90px);
  font-size: 20px;
}

/*
  ========================================
  Calculator
  ========================================
*/

.calculator {
  @extend %grid;
  @extend %m-auto;
  gap: 25px;
  grid-template-rows: 50px auto;
  width: max-content;

  .display {
    @extend %bg-basic;
    text-align: right;
  }

  .keyboard {
    @extend %grid;
    @extend %bg-basic;
    grid-template-columns: repeat(4, 100px);
    grid-template-rows: repeat(5, 50px);
    gap: 10px;

    .span-two-col {
      grid-column: span 2;
    }

    button {
      @extend %border-radius-basic;
      @include basic-border($purple-dark);
      color: inherit;
      font-family: inherit;

      &:active {
        transform: scale(0.95);
      }
      
    }
  }
}
