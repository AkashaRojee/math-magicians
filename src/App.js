/* eslint-disable */

import './App.scss';
import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Header from './components/Header';
import Home from './routes/Home';
import Calculator from './routes/Calculator';
import Quote from './routes/Quote';

const App = () => (
  <BrowserRouter>
    <div className="App">
      <Header />
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/calculator">
          <Calculator />
        </Route>
        <Route exact path="/quote">
          <Quote />
        </Route>
      </Switch>
    </div>
  </BrowserRouter>
);

export default App;
