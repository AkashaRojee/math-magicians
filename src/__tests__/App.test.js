import renderer from 'react-test-renderer';
import App from '../App';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });
});

test('Clicking on Calculator link in navigation bar displays calculator in App', () => {
  render(<App />);
  const calculatorLink = screen.getByText('Calculator');

  userEvent.click(calculatorLink);
  const calculatorButtons = screen.getAllByRole('button');
  
  calculatorButtons.forEach((calculatorButton) => {
    expect(calculatorButton).toBeVisible();
  });
});