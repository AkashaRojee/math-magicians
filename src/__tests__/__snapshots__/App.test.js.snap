// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="header"
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <a
        href="/math-magicians/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/math-magicians/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/math-magicians/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </nav>
  </div>,
  <main
    className="home-route"
  >
    <h2>
      Welcome to our page!
    </h2>
    <div>
      <p>
        Lorem ipsum 1
      </p>
      <p>
        Lorem ipsum 2
      </p>
      <p>
        Lorem ipsum 3
      </p>
    </div>
  </main>,
]
`;
